app/repositories/page-content.tsx

```diff
@@ -90,6 +99,7 @@ const getRepos =
   }
 
 function useSearchResultsQuery(q: string) {
+  const abortController = useRef<AbortController | null>(null)
```

Comment:
I think i was avoiding this earlier on so that folks can run one of these and leave it running in the background while they browse and do other things. but in hindsight, that may be a bit too risky since it will make it very easy for people to leave something like this running for 20K followers or something.

Reply:
agreed on the refactoring into a hook recommendation. I'm adding a ticket for that and don't wanna bite that off in this PR. let me get this out and will handle the refactoring in a future, separate PR.